libcloud/common/aws.py

```diff
@@ -178,6 +180,97 @@ def _get_aws_auth_param(self, params, secret_key, path='/'):
         return b64_hmac.decode('utf-8')
 
 
+class V4SignedAWSConnection(AWSTokenConnection):
+
+    def add_default_params(self, params):
+        params['Version'] = self.version
+        return params
+
+    def pre_connect_hook(self, params, headers):
+        now = datetime.utcnow()
+        headers['X-AMZ-Date'] = now.strftime('%Y%m%dT%H%M%SZ')
+        headers['Authorization'] = \
+            self._get_authorization_v4_header(params, headers, now)
+
+        return params, headers
+
+    def _get_authorization_v4_header(self, params, headers, dt):
+        assert self.method == 'GET', 'AWS Signature V4 not implemented for ' \
+                                     'other methods than GET'
```

Comment:
Please document what a future committer would have to change to get this working. I assume writing code to serialize the POST data?